cloud/scope/machine_pool.go

```diff
@@ -536,27 +596,24 @@ func (m *MachinePoolScope) CreateInstancePool(ctx context.Context) (*core.Instan
 }
 
 // UpdatePool attempts to update the instance pool
-func (m *MachinePoolScope) UpdatePool(ctx context.Context, instancePool *core.InstancePool) error {
-
+func (m *MachinePoolScope) UpdatePool(ctx context.Context, instancePool *core.InstancePool) (*core.InstancePool, error) {
 	if instancePoolNeedsUpdates(m, instancePool) {
-		m.Info("updating instance pool")
-
-		replicas := int(1)
-		if m.MachinePool.Spec.Replicas != nil {
-			replicas = int(*m.MachinePool.Spec.Replicas)
-		}
-
+		m.Info("Updating instance pool")
+		replicas := int(*m.MachinePool.Spec.Replicas)
```

Comment:
Is it possible for `Replicas` at this point to be `nil`?

Reply:
fixed